development-vm/replication/sync-aws-elasticsearch.sh

```diff
@@ -0,0 +1,102 @@
+USAGE_DESCRIPTION="Download Elasticsearch index archives and import into another instance.
+
+If one or more index names are given, only those index files are imported;
+otherwise, all index files are imported."
+. ./common-args.sh
+if $SKIP_ELASTIC; then
+  exit
+fi
+
+shift $(($OPTIND-1))
+
+LOCAL_ES_HOST="http://localhost:9200/"
+LOCAL_ARCHIVE_PATH="${DIR}/elasticsearch"
+
+status "Starting search index replication from AWS"
+
+if $SKIP_DOWNLOAD; then
+  status "Skipping fetch of new archives"
+
+  if [ ! -d $LOCAL_ARCHIVE_PATH ]; then
+    error "Download directory ${LOCAL_ARCHIVE_PATH} does not exist and fetching new archives has been skipped."
+    exit 1
+  fi
+else
+  if [ -d $LOCAL_ARCHIVE_PATH ]; then
+    ok "Download directory ${LOCAL_ARCHIVE_PATH} exists"
+  else
+    status "Creating local backups directory"
+    mkdir -p $LOCAL_ARCHIVE_PATH
+    ok "Download directory ${LOCAL_ARCHIVE_PATH} created"
+  fi
+
+  status "Downloading latest es backups from AWS"
+
+  remote_file_details=$(aws s3 ls s3://govuk-integration-database-backups/elasticsearch/indices/)
+  remote_paths=$(echo $remote_file_details | ruby -e 'STDOUT << STDIN.read.split("PRE").group_by { |n| n.split(/-\d/).first }.map { |_, d| d.sort.last.strip }.join(" ")')
+  for remote_path in $remote_paths; do
+    status "Syncing data from ${remote_path}"
+    mkdir $LOCAL_ARCHIVE_PATH/$remote_path
+    aws s3 sync s3://govuk-integration-database-backups/elasticsearch/indices/${remote_path} $LOCAL_ARCHIVE_PATH/$remote_path/ --only-show-errors
+  done
+
+  exit 0
+fi
+
+exit 1
+FILE_COUNT=`ls -l ${LOCAL_ARCHIVE_PATH}/*.zip | wc -l | tr -d ' '`
+
+if [ $FILE_COUNT -lt 1 ]; then
+  error "No archives found in ${LOCAL_ARCHIVE_PATH}"
+  exit 1
+elif [ $FILE_COUNT -eq 1 ]; then
+  ok "${FILE_COUNT} archive found"
+else
+  ok "${FILE_COUNT} archives found"
+fi
+
+if [ $# -gt 0 ]; then
+  filenames=""
+  for index_name in $@; do
+    filenames="$filenames $LOCAL_ARCHIVE_PATH/$index_name.zip"
+  done
+
+  for filename in $filenames; do
+    if [ ! -e $filename ]; then
+      error "File $filename not found: aborting."
+      exit 1
+    fi
+  done
+else
+  filenames=${LOCAL_ARCHIVE_PATH}/*.zip
+fi
+
+status "Restoring data into Elasticsearch"
+
+for f in $filenames
+do
+  if $DRY_RUN; then
+    status "$f (dry run)"
+  else
+    if curl $LOCAL_ES_HOST -o /dev/null 2> /dev/null; then
+      ok "Elasticsearch is running on ${LOCAL_ES_HOST}"
+    else
+      error "Elasticsearch is not running on ${LOCAL_ES_HOST}. Aborting..."
+      exit 1
+    fi
+
+    alias_name=$(basename $f .zip)
+    iso_date="$(date --iso-8601=seconds|cut --byte=-19|tr [:upper:] [:lower:])z"
+    real_name="$alias_name-$iso_date-00000000-0000-0000-0000-000000000000"
+
+    status "Importing $f to $alias_name (real name $real_name)"
+    bundle exec es_dump_restore restore_alias "$LOCAL_ES_HOST" "$alias_name" "$real_name" "$f" '{"settings":{"index":{"number_of_replicas":"0","number_of_shards":"1"}}}' 250
```

Comment:
Are you planning to update this before we merge this PR? Otherwise we should probably move this commit to another PR.

Reply:
Sorry, ignore this comment. I've been reviewing commit-by-commit and forgot to check the later ones. 🤦‍♀️

Reply:
done